Replicator/tests/ReplicatorLoopbackTest.hh

```diff
@@ -261,6 +263,9 @@ public:
             {
                 _statusReceived = status;
                 _checkStopWhenIdle();
+                if (_statusReceived.level == kC4Idle && _callbackWhenIdle) {
```

Comment:
I think it should be called before _checkStopWhenIdle(), otherwise, there is a case that it will be called after the replicator is called to stop.

Reply:
_checkStopWhenIdle() does not stop unless the caller (test case) requests with stopWhenIdle(). The caller is responsible for synchronizing between stopWhenIdle and _callbackWhenIdle. If both are requested, It's kind of UD (undefined behavior). If do callback first, the state is still Idle, and we will stop. That's usually not we wanted. In my tests, I request StopWhenIdle() inside the callback and then nullify _callbackWhenIdle.